lib/LedgerSMB/PSGI.pm

```diff
@@ -161,8 +161,14 @@ sub psgi_app {
         }
 
         $LedgerSMB::App_State::DBH = $request->{dbh};
-        ($status, $headers, $body) = @{&$action($request)};
-
+        if ( defined $action ) {
```

Comment:
The cause of the failure is a bit deeper: even though there is no database, something tries to access the database. I identified that to be Template.pm. See my PR.